libshvbroker/src/rpc/clientconnectiononbroker.cpp

```diff
@@ -298,6 +298,9 @@ void ClientConnectionOnBroker::processLoginPhase()
 	}
 	Super::processLoginPhase();
 	BrokerApp::instance()->checkLogin(m_userLoginContext, this, [this] (auto result) {
+		if (result.userNameOverride) {
+			m_userLogin.user = *result.userNameOverride;
```

Comment:
I'm not a big fan of this `std::optional` override, because it cannot be distinguished from pointer operation without knowing `UserLoginResult` definition. Is it possible to use `m_userLogin.user = result.userNameOverride.value();`?

Reply:
>it cannot be distinguished from pointer operation without knowing UserLoginResult definition

I don't think this is such an issue. The semantics of the types (optional/pointer) are similar.

>Is it possible to use m_userLogin.user = result.userNameOverride.value();?

Yes. I should stop using the asterisk altogether and always use .value(). Fixed.